src/main/java/dev/dkvl/womutils/panel/WomPanel.java

```diff
@@ -115,111 +120,30 @@ public void mouseClicked(MouseEvent e)
 
         c.gridy++;
 
-        JPanel buttonsPanel = new JPanel();
-        buttonsPanel.setLayout(new GridBagLayout());
-        buttonsPanel.setBorder(new EmptyBorder(0, 0, 10, 0));
-
-        GridBagConstraints gbc = new GridBagConstraints();
-        gbc.fill = GridBagConstraints.HORIZONTAL;
-        gbc.gridx = 0;
-        gbc.weightx = 1;
-        gbc.ipady = 10;
-
-        JButton updateButton = new JButton();
-        updateButton.setFont(FontManager.getRunescapeSmallFont());
-        updateButton.setEnabled(false);
-        updateButton.addActionListener(e ->
-            womClient.updatePlayer(sanitize(searchBar.getText())));
-        updateButton.setText("Update");
-
-        JButton profileButton = new JButton();
-        profileButton.setFont(FontManager.getRunescapeSmallFont());
-        profileButton.setEnabled(false);
-        profileButton.addActionListener(e ->
-            openPlayerProfile(sanitize(searchBar.getText())));
-        profileButton.setText("Open Profile");
-
-        buttons.add(updateButton);
-        buttons.add(profileButton);
-
-        buttonsPanel.add(updateButton, gbc);
-        gbc.gridx++;
-        gbc.insets.left = 7;
-        buttonsPanel.add(profileButton, gbc);
-
-        add(buttonsPanel, c);
+        add(createButtonsPanel(), c);
 
         c.gridy++;
 
+        MiscInfo lastUpdated = MiscInfo.LAST_UPDATED;
+        miscInfoLabels.add(MiscInfo.BUILD);
+        miscInfoLabels.add(MiscInfo.COUNTRY);
+        miscInfoLabels.add(MiscInfo.TTM);
+        miscInfoLabels.add(MiscInfo.EHP);
+        miscInfoLabels.add(MiscInfo.EHB);
+        miscInfoLabels.add(MiscInfo.EXP);
+        miscInfoLabels.add(lastUpdated);
```

Comment:
Can you instantiate the labels here through some wrapper class in a loop over the MiscInfo enum, then those would be the labels we write to.